storage/cloudspanner/map_storage_test.go

```diff
@@ -42,12 +58,68 @@ func GetTestDB(ctx context.Context, t *testing.T) *spanner.Client {
 	return client
 }
 
+// Unique per test binary invocation
+var timestamp = time.Now().UTC().Format("jan-02-15-04-05")
+var testBinary = strings.ToLower(strings.Replace(path.Base(os.Args[0]), ".test", "", 1))
+var invocationID = fmt.Sprintf("%s-%s", timestamp, testBinary)
+var dbCount uint32 // Unique per test invocation
+
+func uniqueDBName(project, instance string) string {
+	database := fmt.Sprintf("%s-%d", invocationID, atomic.AddUint32(&dbCount, 1))
+	return fmt.Sprintf("projects/%s/instances/%s/databases/%s", project, instance, database)
+}
+
+func inMemClient(ctx context.Context, t testing.TB, dbName string, statements []string) *spanner.Client {
+	t.Helper()
+	// Don't use SPANNER_EMULATOR_HOST because we need the raw connection for
+	// the database admin client anyway.
+
+	t.Logf("Using in-memory fake Spanner DB: %s", dbName)
+	srv, err := spannertest.NewServer("localhost:0")
+	if err != nil {
+		t.Fatalf("Starting in-memory fake: %v", err)
+	}
+	t.Cleanup(srv.Close)
+	srv.SetLogger(t.Logf)
+	dialCtx, cancel := context.WithTimeout(ctx, 1*time.Second)
+	defer cancel()
+	conn, err := grpc.DialContext(dialCtx, srv.Addr, grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("Dialing in-memory fake: %v", err)
+	}
+	client, err := spanner.NewClient(ctx, dbName, option.WithGRPCConn(conn))
+	if err != nil {
+		conn.Close()
+		t.Fatalf("Connecting to in-memory fake: %v", err)
+	}
+	t.Cleanup(client.Close)
+
+	// Set database schema
+	t.Logf("DDL update: %s", statements)
+	adminClient, err := database.NewDatabaseAdminClient(ctx, option.WithGRPCConn(conn))
+	if err != nil {
+		t.Fatalf("Connecting to in-memory fake DB admin: %v", err)
+	}
+	op, err := adminClient.UpdateDatabaseDdl(ctx, &databasepb.UpdateDatabaseDdlRequest{
+		Database:   dbName,
+		Statements: statements,
+	})
+	if err != nil {
+		t.Fatalf("Starting DDL update: %v", err)
+	}
+	if err := op.Wait(ctx); err != nil {
+		t.Fatal(err)
```

Comment:
Maybe `Fatalf("Failed to update DB: %v", err)`?

Reply:
done